zul/src/archive/web/js/zul/wgt/Toolbar.js

```diff
@@ -50,7 +50,55 @@ zul.wgt.Toolbar = zk.$extends(zul.Widget, {
 		/** Sets the orient.
 		 * @param String orient either "horizontal" or "vertical".
 		 */
-		orient: _zkf
+		orient: _zkf,
+		/**
+		 * Return whether toolbar has a button that shows a popup
+		 * which contains those content weren't able to fit in the toolbar.
+		 * If overflowPopup is false, toolbar will display multiple rows when content is wider than toolbar.
+		 * Default: false.
+		 *
+		 * @return boolean
+		 * @since 8.6.0
+		 */
+		 /**
+		 * Set whether toolbar has a button that shows a popup
+		 * which contains those content weren't able to fit in the toolbar.
+		 * If overflowPopup is false, toolbar will display multiple rows when content is wider than toolbar.
+		 *
+		 * @param boolean overflowPopup whether toolbar has a button that shows a popup
+		 * @since 8.6.0
+		 */
+		overflowPopup: function () {
+			this.rerender();
+		}
+	},
+	bind_: function () {
+		this.$supers('bind_', arguments);
+		if (this.isOverflowPopup()) {
+			zWatch.listen({onSize: this});
+			zWatch.listen({onCommandReady: this});
+			this.domListen_(this.$n('overflowpopup-button'), 'onClick', '_openPopup');
+			this._initPopup();
+		}
+	},
+	unbind_: function () {
+		var popup = this._pp;
+		if (popup) {
+			popup.detach();
+			this.domUnlisten_(this.$n('overflowpopup-button'), 'onClick', '_openPopup');
+			zWatch.unlisten({onCommandReady: this});
+			zWatch.unlisten({onSize: this});
+		}
+		this.$supers('unbind_', arguments);
+	},
+	_initPopup: function () {
+		var popup = this._pp = new zul.wgt.Popup();
```

Comment:
Just use div to build a popup.
There is no need to use Popup widget.

You need to sync position manually